test/teos/unit/test_carrier.py

```diff
@@ -1,38 +1,42 @@
-from bitcoind_mock.transaction import create_dummy_transaction
-from test.teos.unit.conftest import generate_blocks, get_random_value_hex
+from test.teos.conftest import generate_blocks
+from test.teos.unit.conftest import get_random_value_hex
 from teos.rpc_errors import RPC_VERIFY_ALREADY_IN_CHAIN, RPC_DESERIALIZATION_ERROR
 
+from test.teos.conftest import create_commitment_tx, bitcoin_cli
+
 
 # FIXME: This test do not fully cover the carrier since the simulator does not support every single error bitcoind may
 #        return for RPC_VERIFY_REJECTED and RPC_VERIFY_ERROR. Further development of the simulator / mocks or simulation
 #        with bitcoind is required
+# TODO: Given we are getting rid of the mock, test the uncovered cases.
```

Comment:
TODO here, maybe open a new issue for it?

Reply:
Just opened one https://github.com/talaia-labs/python-teos/issues/184